### PR TITLE
feat: disable features

### PR DESCRIPTION
**Pull request by alaister**
## What kind of change does this PR introduce?

feature

## What is the new behaviour?

Features can be disabled as needed

### CODE REVIEW

**Comment by joshenlim** [comment]
probably need to add billing:all check to `<UpgradeToPro>` as well?

File: studio/__mocks__/hooks.js [hunk 1]
```diff
@@ -8,6 +8,8 @@ export const useParams = jest.fn().mockImplementation(() => ({ ref: '123' }))
 
 export const useLocalStorage = jest.fn().mockImplementation(() => [[], jest.fn()])
 
+export const useIsFeatureEnabled = jest.fn().mockImplementation(() => [])
```
Comment (kevcodez) [inline studio/__mocks__/hooks.js:11]: Should we actually check the arg and return `true` if a string is passed in an `{}` if an array is passed in? Never returns a boolean does it?

File: studio/data/projects/project-restore-mutation.ts [hunk 2]
```diff
@@ -11,7 +11,6 @@ export type ProjectRestoreVariables = {
 export async function restoreProject({ ref }: ProjectRestoreVariables) {
   const { data, error } = await post('/platform/projects/{ref}/restore', {
     params: { path: { ref } },
```
Comment (kevcodez) [inline studio/data/projects/project-restore-mutation.ts:13]: Just double-checking this doesn't cause any issues - sometimes fetch without a body can cause weird issues

File: studio/hooks/misc/useIsFeatureEnabled.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { Feature } from 'data/profile/types'
+import { useProfile } from 'lib/profile'
+
+function checkFeature(feature: Feature, features?: Feature[]) {
+  return !features?.includes(feature) ?? true
+}
+
+type SnakeToCamelCase<S extends string> = S extends `${infer First}_${infer Rest}`
+  ? `${First}${SnakeToCamelCase<Capitalize<Rest>>}`
```
Comment (kevcodez) [inline studio/hooks/misc/useIsFeatureEnabled.ts:9]: Proud of ya son

**Review by ivasilov — APPROVED** [review]

File: studio/__mocks__/hooks.js (hunk 1 above)
Comment (alaister) [reply studio/__mocks__/hooks.js:11]: True – good catch – I've updated!

**Review by joshenlim — APPROVED** [review]

**Pull request merged**